File: tests/test_cpg/fixtures/parity_pt.py
"""Cross-language path traversal fixture — Python."""

import os


def read_file():
    filename = os.environ.get("FILE_PATH")  # $ source=path_traversal
    with open(filename) as f:  # $ sink=path_traversal
        return f.read()


def serve_static():
    path = "/tmp/" + os.environ.get("USER_FILE", "")  # $ source=path_traversal
    return open(path).read()  # $ sink=path_traversal

File: tests/test_cpg/fixtures/parity_pt.java
/** Cross-language path traversal fixture — Java. */

import java.io.FileInputStream;

public class FileService {

    public void readFile() {
        String filename = System.getenv("FILE_PATH");  // $ source=path_traversal
        new FileInputStream(filename);  // $ sink=path_traversal
    }

    public void serveStatic() {
        String p = "/tmp/" + System.getenv("USER_FILE");  // $ source=path_traversal
        new FileInputStream(p);  // $ sink=path_traversal
    }
}
